sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java

```diff
@@ -37,7 +37,7 @@
                 request.getResourceType(),
                 request.getOperationType());
 
-        AuthorizationTokenType originalRequestTokenType = request.authorizationTokenType;
+        AuthorizationTokenType originalRequestTokenType = authorizationTokenProvider.getAuthorizationTokenType();
 
```

Comment:
do we expect the original request and the barrier request to have different auth token types?

why can't we rely on the request.authTokenType?

Reply:
In regular flow we should be able to extract the auth type from the current request. Yet, if we do have a valid authorization token provider with a valid authorization token type, we should honor that rather than the one in the request object passed in which might come with some rather default setting.

In the case of the AAD token, since it does get automatically refreshed in the background, we do want to make sure we reflect the latest value and not reuse some value that might not be current (an unlikely scenario since token refresh is at least 10 minutes before expiry but not impossible).

Reply:
> In the case of the AAD token, since it does get automatically refreshed in the background, we do want to make sure we reflect the latest value and not reuse some value that might not be current (an unlikely scenario since token refresh is at least 10 minutes before expiry but not impossible).

the background refresh should only refresh the value. correct? not the token type?
why token type should change during a refresh?

Reply:
do we expect the token type and authorization type to change in the lifetime of a single Cosmos Client?

Reply:
@moderakh we should try to keep the conversation in one place please. 
The token type does not change between the requests except if the request was created with some default value in the first place and not with the token type matching the one in the CosmosClient.
Is the question: why we do have requests that might not use the same value we set in the CosmosClient initialization? We should not except that there seem to be some tests that do so.

Reply:
the explanation makes sense. thanks @milismsft